flatliners/prometheusendpoint.py

```diff
@@ -0,0 +1,51 @@
+from time import sleep, time
+from collections import defaultdict
+from .baseflatliner import BaseFlatliner
+
+from prometheus_client import Gauge, start_http_server
+
+# Set up logging
+import logging
+_LOGGER = logging.getLogger(__name__)
+
+class PrometheusEndpoint(BaseFlatliner):
+    def __init__(self):
+        super().__init__()
+        self.pruning_interval = 5*60 # seconds # This value should be close to the scraping interval for your prometheus
+        _LOGGER.info("Prometheus Endpoint initialized. Metric pruning interval is {0} seconds".format(self.pruning_interval))
+        self.published_metric_timestamps = defaultdict(list)
+        self.weirdness_score_gauge = Gauge('weirdness_score','Weirdness score for the given Cluster and Version',['cluster','version'])
+
+    def on_next(self, x):
+        try:
+            self.weirdness_score_gauge.labels(cluster=str(x.cluster), version=str(x.version)).set(x.weirdness_score)
+            self.published_metric_timestamps[str(x.cluster)] = [int(time()),str(x.version)] # timestamp when the metric was published and metric version info
+        except Exception as e:
+            _LOGGER.error("Couldn't process the following packet {0}. Reason: {1}".format(x,str(e)))
+            raise e
+        pass
+
+    def _delete_prev_metrics(self):
+        '''
+        This function will remove any metric that was published 3 scraping intervals ago or older
+        '''
+        current_timestamp = int(time())
+        for cluster_id in list(self.published_metric_timestamps):
+            if (current_timestamp - self.published_metric_timestamps[cluster_id][0]) > (3 * self.pruning_interval):
```

Comment:
let's extract a `timestamp_threshold` so that you dont have to calculate in every loop cycle.

```python
timestamp_threshold = current_timestamp - 3 * self.pruning_interval
if self.published_metric_timestamps[cluster_id][0] < timestamp_threshold:
```

Reply:
good idea